### PR TITLE
Anchor benchmark on commit

### PR DESCRIPTION
**Pull request by Alan32Liu**
Background: #1632

Fix coverage-based benchmarks:
1. Remove `--depth 1` when cloning the benchmark repo.
2. Add the commit hash to the benchmark's `benchmark.yaml`.
3. Add the commit date to the benchmark's `benchmark.yaml`.

### CODE REVIEW

File: benchmarks/libxslt_xpath/Dockerfile [hunk 1]
```diff
@@ -58,12 +58,9 @@ RUN apt-get update && \
 
 
 RUN git clone \
-        --branch v2.10.3 \
-        --depth 1 \
         https://gitlab.gnome.org/GNOME/libxml2.git && \
+    git -C libxml2 checkout c7260a47f19e01f4f663b6a56fbdc2dafd8a6e7e
```
Comment (jonathanmetzman) [inline benchmarks/libxslt_xpath/Dockerfile:62]: It's weird we are going to checkout one commit here and then another in benchmark.yaml. Can we do without the checkout here?

**Review by jonathanmetzman — APPROVED** [review]
lgtm with nits!
Thanks!

File: benchmarks/libxslt_xpath/Dockerfile (hunk 1 above)
Comment (Alan32Liu) [reply benchmarks/libxslt_xpath/Dockerfile:62]: IIUC, `libxslt` is the benchmark target and `libxml2` is its dependency?
This commit ID is for `libxml2` and the commit ID in `benchmark.yaml` is for `libxslt`.

**Comment by Alan32Liu** [comment]
> lgtm with nits! Thanks!

Thanks!
Before merging this, I also plan to unify their names to `<project_name>_<target_name>`.
This should make the names easier to understand and less misleading (e.g., `2017` in `freetype2-2017`, `1.3.2` in `harfbuzz-1.3.2` might be outdated). Their versions can be identified by the commit IDs in the `benchmark.yaml`s.

File: benchmarks/libxslt_xpath/Dockerfile (hunk 1 above)
Comment (jonathanmetzman) [reply benchmarks/libxslt_xpath/Dockerfile:62]: Ah my bad.

**Review by jonathanmetzman — APPROVED** [review]
lgtm

**Comment by Alan32Liu** [comment]
/gcbrun run_experiment.py --experiment-config /opt/fuzzbench/service/experiment-config.yaml --benchmarks bloaty_fuzz_target freetype2_ftfuzzer harfbuzz_hb-shape-fuzzer jsoncpp_jsoncpp_fuzzer libjpeg-turbo_libjpeg_turbo_fuzzer libpng_libpng_read_fuzzer libxml2-v2.9.2 libxslt_xpath mbedtls_fuzz_dtlsclient openh264_decoder_fuzzer openthread_ip6-send-fuzzer   --experiment-name gh-experiment-2 --fuzzers afl libfuzzer centipede honggfuzz

**Comment by jonathanmetzman** [comment]
Crap sorry I deleted gh-experiment-2

**Comment by jonathanmetzman** [comment]
/gcbrun run_experiment.py --experiment-config /opt/fuzzbench/service/experiment-config.yaml --benchmarks bloaty_fuzz_target freetype2_ftfuzzer harfbuzz_hb-shape-fuzzer jsoncpp_jsoncpp_fuzzer libjpeg-turbo_libjpeg_turbo_fuzzer libpng_libpng_read_fuzzer libxml2-v2.9.2 libxslt_xpath mbedtls_fuzz_dtlsclient openh264_decoder_fuzzer openthread_ip6-send-fuzzer --experiment-name gh-experiment-3 --fuzzers afl libfuzzer centipede honggfuzz

**Comment by Alan32Liu** [comment]
> gh-experiment-2

Sorry, what is `gh-experiment-2`?

**Comment by Alan32Liu** [comment]
This is the second time we have had CI failures due to https://gitlab.gnome.org/GNOME being down and we cannot clone repos from it.
But I presume there is no easy way to avoid/mitigate this? (e.g. mirroring their repos, etc.)

**Pull request merged**